apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts

```diff
@@ -113,81 +120,103 @@ export class SendMessagePush extends SendMessageBase {
-      await this.sendErrors(command);
+      await this.sendNoActiveChannelError(command.job);
+      await this.sendNotificationError(command.job);
 
       return;
     }
 
-    for (const channel of pushChannels) {
-      if (!channel.credentials?.deviceTokens) {
-        await this.createExecutionDetails.execute(
-          CreateExecutionDetailsCommand.create({
-            ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
-            detail: DetailEnum.PUSH_MISSING_DEVICE_TOKENS,
-            source: ExecutionDetailsSourceEnum.INTERNAL,
-            status: ExecutionDetailsStatusEnum.FAILED,
-            isTest: false,
-            isRetry: false,
-          })
-        );
-
-        continue;
-      }
+    const messagePayload = Object.assign({}, command.payload);
+    delete messagePayload.attachments;
 
-      const integration = await this.getIntegration({
-        id: channel._integrationId,
-        organizationId: command.organizationId,
-        environmentId: command.environmentId,
-        channelType: ChannelTypeEnum.PUSH,
-        providerId: channel.providerId,
-        userId: command.userId,
-        filterData: {
-          tenant: command.job.tenant,
-        },
-      });
+    let integrationsWithErrors = 0;
+    for (const channel of pushChannels) {
+      const { deviceTokens } = channel.credentials || {};
 
-      if (!integration) {
-        await this.createExecutionDetails.execute(
-          CreateExecutionDetailsCommand.create({
-            ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
-            detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION,
-            source: ExecutionDetailsSourceEnum.INTERNAL,
-            status: ExecutionDetailsStatusEnum.FAILED,
-            isTest: false,
-            isRetry: false,
-          })
-        );
+      const [isChannelMissingDeviceTokens, integration] = await Promise.all([
+        this.isChannelMissingDeviceTokens(channel, command),
+        this.getSubscriberIntegration(channel, command),
+      ]);
 
+      // We avoid to send a message if subscriber has not an integration or if the subscriber has no device tokens for said integration
+      if (!deviceTokens || !integration || isChannelMissingDeviceTokens) {
+        integrationsWithErrors++;
         continue;
       }
 
       await this.sendSelectedIntegrationExecution(command.job, integration);
 
       const overrides = command.overrides[integration.providerId] || {};
 
-      await this.sendMessage(
+      const result = await this.sendMessage(
         subscriber,
         integration,
-        channel.credentials.deviceTokens,
+        deviceTokens,
         title,
         content,
         command,
         command.payload,
         overrides,
         stepData
       );
+
+      if (!result) {
+        integrationsWithErrors++;
+      }
+    }
+
+    if (integrationsWithErrors > 0) {
+      Logger.error(
+        { jobId: command.jobId },
```

Comment:
```suggestion
        { jobId: command.jobId ,
        errorCount: integrationsWithErrors},
```
Should we add the count as well?

Reply:
Will it be helpful the count number? I added this more thinking in a potential NR dashboard monitoring the failures in the Push channel notifications. For debugging goals the execution details logs should be enough unless I missed any case!

Reply:
no, i think it should be fine, I was not sure if needed or not. we could add it in the future if we find it helpful.